pkgs/development/python-modules/opentsne/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, cython
+, numpy
+, scipy
+, scikit-learn
+, pytestCheckHook
+, nix-update-script
+, opentsne
+}:
+
+buildPythonPackage rec {
+  pname = "opentsne";
+  version = "1.0.0";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "pavlin-policar";
+    repo = "openTSNE";
+    rev = "v${version}";
+    hash = "sha256-L5Qx6dMJlXF3EaWwlFTQ3dkhGXc5PvQBXYJo+QO+Hxc=";
+  };
+
+  propagatedBuildInputs = [ numpy scipy scikit-learn ];
+  nativeBuildInputs = [ cython ];
+
+  pythonImportsCheck = [ "openTSNE" ];
+  dontCheck = true;
+
+  passthru = {
+    updateScript = nix-update-script {};
+    tests.pytest = buildPythonPackage {
```

Comment:
Can't we run them inline?

Reply:
That's how I started this package but there was a error about cyclic imports and I only got it working when I moved that part to another derivation removing all but the tests passing the library as dependency for it to find the library.

Reply:
Please see https://github.com/NixOS/nixpkgs/blob/staging/pkgs/development/python-modules/constantly/default.nix for an option how I normally do it